modules/ipi-install-configuring-host-network-interfaces-in-the-install-config.yaml-file.adoc

```diff
@@ -0,0 +1,56 @@
+// This is included in the following assemblies:
+//
+// installing_bare_metal_ipi/ipi-install-installation-workflow.adoc
+
+[id="configuring-host-network-interfaces-in-the-install-config.yaml-file_{context}"]
+= (Optional) Configuring host network interfaces in the `install-config.yaml` file
+
+During installation, you can set the `networkConfig` configuration setting in the `install-config.yaml` file to configure host network interfaces using NMState on the `baremetal` network. To use the `networkConfig` configuration setting, you must pass an NMState key file as YAML. See link:https://nmstate.io/examples.html#interfaces-ethernet[NMState] for additional examples of the NMState syntax.
```

Comment:
* You can configure any interface, it doesn't have to be on the `baremetal` network
* A "key file" is the connection file used to configure NetworkManager - i.e. it's the _output_ of NMState. The input is the YAML described here. "NMState keyfile as YAML" is mixing these two concepts and is confusing.

Reply:
@zaneb But do we support it? Some of the meeting notes involved issues like deploying their own DHCP server on the provisioning network, and then the discussion became about whether or not we would support that sort of thing on the provisioning network. I know it will work, but the reason it's constrained the baremetal network is really a question of whether or not we support it. If so, we probably need to describe the managed|unmanaged bits too.

Reply:
Fair question, I don't know what we've decided to support.

Reply:
My thinking when we discussed this previously is that it doesn't make sense to configure the provisioning network this way. You still need to have DHCP running in some capacity to PXE boot that interface. I guess you could statically configure an IP on it after deployment, but I'm not sure what the point of that would be.

I guess we could say that configuration of any interface is supported, but that DHCP still must be allowed on the provisioning NIC if you're using that deployment method? I assume anyone who objects to DHCP on their network is deploying without a provisioning network anyway.

Reply:
You could configure e.g. a storage network, couldn't you? It just seems like there's any number of things you can configure that are nothing to do with either of the networks that we require you to set up.

Reply:
True, whether we support changing the provisioning nic or not, this doesn't need to be limited to the baremetal nic.

Reply:
I removed the qualifying language regarding the baremetal network last week, so I think this is good to go. I can make a statement about supporting other networks if you like.